### PR TITLE
[8.0] Stock Operation Type Location

### PR DESCRIPTION
**Pull request by mikevhe18**
Changelog : 

- Fixing issue when allowed source/destination location has only one data and the same as default source/destination location

- Adds the possibility to filter location of transfer wizard using the same behaviour

### CODE REVIEW

File: stock_operation_type_location/__openerp__.py [hunk 1]
```diff
@@ -4,13 +4,15 @@
 
 {
     'name': 'Stock Operation Type Location',
-    'version': '8.0.1.0.0',
+    'version': '8.0.1.1.1',
```
Comment (andhit-r) [inline stock_operation_type_location/__openerp__.py]: 8.0.1.1.0 enough

**Review by andhit-r — CHANGES_REQUESTED** [review]
Code Review: minor comments inline. Coverage decreased but nothing related to this PR
Functional Review: Ok. Tested on runbot

**Comment by andhit-r** [comment]
@azmimr67 @m-sani please review

**Comment by m-sani** [comment]
tested :+1:

**Comment by mikevhe18** [comment]
Hello @gurneyalex , @jgrandguillaume , @rousseldenis, @pedrobaeza 
Would you kindly review this PR ?

Thanks in Advance

File: stock_operation_type_location/models/stock_move.py [hunk 2]
```diff
@@ -22,21 +22,20 @@ def onchange_picking_type_id(self):
 
             allowed_location = picking_type.allowed_location_ids
             allowed_dest_location = picking_type.allowed_dest_location_ids
-
-            scr_id = picking_type.default_location_src_id
-            dest_id = picking_type.default_location_dest_id
+            scr_id = picking_type.default_location_src_id.id
```
Comment (gurneyalex) [inline stock_operation_type_location/models/stock_move.py:25]: can we take the opportunity of this PR to fix the typo and `s/scr_id/src_id/` ?

File: stock_operation_type_location/wizard/stock_transfer_details.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                domain_destinationloc = "[('id', 'in', %s)]" % dest_loc
+            elif loc and not dest_loc:
+                domain_sourceloc = "[('id', 'in', %s)]" % loc
+                domain_destinationloc = "[]"
+            elif not loc and dest_loc:
+                domain_sourceloc = "[]"
+                domain_destinationloc = "[('id', 'in', %s)]" % dest_loc
+            else:
+                domain_sourceloc = "[]"
+                domain_destinationloc = "[]"
```
Comment (gurneyalex) [inline stock_operation_type_location/wizard/stock_transfer_details.py]: you could much simplify this block by first initializaing domain_sourceloc and domain_destinationloc to "[]" and having two blocks changing the value `if loc:` and `if dest_loc:`

File: stock_operation_type_location/wizard/stock_transfer_details.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    _inherit = "stock.transfer_details"
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form',
+                        toolbar=False, submenu=False):
+        res = super(StockTransferDetails, self)\
+            .fields_view_get(
+                view_id=view_id, view_type=view_type,
+                toolbar=toolbar, submenu=submenu)
+        context = self._context
```
Comment (rousseldenis) [inline stock_operation_type_location/wizard/stock_transfer_details.py]: Should use self.env.context

**Review by rousseldenis — COMMENTED** [review]
@mikevhe18 Tested on runbot ok. Minor comment

**Comment by mikevhe18** [comment]
Fixed. Thanks @gurneyalex @rousseldenis.

**Review by rousseldenis — APPROVED** [review]
:+1

**Comment by azmimr67** [comment]
Has been tested on runboot :+1:

**Review by gurneyalex — APPROVED** [review]

**Pull request merged**